### PR TITLE
Feat/uads 18/front end home page

### PR DESCRIPTION
**Pull request by khir829**
# Ticket

https://uads.atlassian.net/browse/UADS-18

# Description

- Implemented the Home Page
- Responsive for Desktop, iPad and Mobile
- NOTE: Data is hardcoded for now as CMS is not integrated yet

# Tests

- Has not been tested

# Images

Web View
image
image

iPad View
image
image
image

Mobile View
image
image
image

### CODE REVIEW

File: web/src/pages/Home.tsx [hunk 1]
```diff
@@ -1,10 +1,231 @@
...
+          <div className=" justify-start">
+            <h2 className="text-center font-bold">Tsujiri</h2>
+            <SponsorsButton />
+          </div>
+          <div className="flex flex-col justify-between content-start mx-10"></div>
+        </div>
+      </div>
+    </>
+  );
+}
```
Comment (khir829) [inline web/src/pages/Home.tsx]: Can just reuse one SponsorCard component. We can delete SponsorCard1 and SponsorCard2.

File: web/src/pages/Home.tsx [hunk 2]
```diff
@@ -1,10 +1,231 @@
...
+
+interface SponsorCardProps {
+  image: string;
+}
+
+function SponsorCard({ image }: SponsorCardProps) {
```
Comment (khir829) [inline web/src/pages/Home.tsx]: I've added the image as an example prop. Try now passing in the shop name text as a parameter. Hint: use the interface SponsorCardProps to determine the type of the parameter

File: web/src/pages/Home.tsx [hunk 3]
```diff
@@ -1,10 +1,231 @@
...
+        <img
+          className="max-w-full overflow-hidden rounded-3xl justify-self-start"
+          src={tsujiri}
+        ></img>
+        <div className="flex flex-row">
+          <div className=" justify-start">
+            <h2 className="text-center font-bold">Tsujiri</h2>
+            <SponsorsButton />
+          </div>
+          <div className="flex flex-col justify-between content-start mx-10"></div>
```
Comment (khir829) [inline web/src/pages/Home.tsx:105]: delete unused div code

File: web/src/pages/Home.tsx [hunk 4]
```diff
@@ -1,10 +1,231 @@
...
+      <div className="flex flex-row flex-nowrap justify-center gap-5 m-10">
+        <EventCard />
+        <EventCard />
+        <EventCard />
+      </div>
+    </>
+  );
+}
+
+function EventCard() {
```
Comment (khir829) [inline web/src/pages/Home.tsx]: Can use props to pass in information (e.g event title, event date). Lookup online or try looking at the SponsorCard or PinkButton component on how to do this.

File: web/src/pages/Home.tsx (hunk 3 above)
Comment (khir829) [inline web/src/pages/Home.tsx]: delete unused code `<div className="flex flex-col justify-between content-start mx-10"></div>`

File: web/src/pages/Home.tsx (hunk 4 above)
Comment (n1cholastan) [reply web/src/pages/Home.tsx]: added props for all relevant elements

**Comment by khir829** [comment]
Socials have yet to be added but can be added at a later stage as it is done on the About page branch

**Review by Harsheel12 — APPROVED** [review]
Looks great! Good job

**Pull request merged**